common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -405,9 +405,13 @@ private void launchCompanyPortal() {
     private void processAmazonAppUri(@NonNull final String url) {
         final String methodTag = TAG + ":processAmazonAppUri";
 
-        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
-        getActivity().startActivity(intent);
-        Logger.info(methodTag, "Sent Intent to launch Amazon app");
+        try {
+            final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+            getActivity().startActivity(intent);
+            Logger.info(methodTag, "Sent Intent to launch Amazon app");
+        } catch (final Throwable throwable) {
+            Logger.error(methodTag, "Failed to launch Amazon app.", e);
```

Comment:
@shahzaibj should we properly handle the error and return to the caller as a MsalClientException?

Reply:
> would the user get stuck  

Actually the opposite is the case, because the website has now the chance to render an appropriate hint to the user, that he should install the app. Current behavior is, that it fails immediately without an useful error message.

Furthermore, this implementation continues the approach that is already used in the other cases:

see here:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/1bb6050d52e775c71b17c339243511c7139098bc/common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java#L367-L374

and here:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/1bb6050d52e775c71b17c339243511c7139098bc/common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java#L379-L388

and here:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/1bb6050d52e775c71b17c339243511c7139098bc/common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java#L336-L341